### PR TITLE
My enumerable

### PR DESCRIPTION
**Pull request by Debas-31**
- [x] Created class `MyList `
- [x] A module `MyEnumerabl`
- [x] Implemented [description of #all? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-all-3F)
- [x] Implemented [description of #any? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-any-3F)
- [x] Implemented [description of #filter method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-filter)

### CODE REVIEW

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Debas-31,** 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## **Invalid Code Review Request**

You have submitted a project, that _**Linters are not passing, please add linters.**_

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+list = MyList.new(1, 2, 3, 4)
+
+p(list.all? { |e| e < 5 })
+p(list.all? { |e| e > 5 })
+
+p(list.any? { |e| e == 2 })
+p(list.any? { |e| e == 5 })
+
+p(list.filter(&:even?))
```
Comment (arslanbisharat) [inline my_list.rb:24]: - [ ] Make sure you have implemented the` #each` method that yields all members of `@list `in succession.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: my_list.rb (hunk 1 above)
Comment (Nuri1977) [reply my_list.rb:18]: .Create a module My[Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html) that implements the following methods (they should have the same funcionality as methods in Enumerable):
#all?
#any?
#filter





[4:47](https://microverse-students.slack.com/archives/D039JAVCKFD/p1648738038074989)
each is not in the requirements

Comment (matovu-farid) [reply my_list.rb:24]: Please elaborate on this.
The #each method is not part of the project requirements

@arslanbisharat

**Review by EricMbouwe — APPROVED** [review]
### Hi @Debas-31, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (arslanbisharat) [reply my_list.rb:24]: Please check the project requirements. You would find that there.

**Pull request merged**